### PR TITLE
Use new MazeRunner /uploads endpoint

### PR DESCRIPTION
**Pull request by twometresteve**
## Goal

Updates the e2e test configuration and scenarios to use the newly introduced "/uploads" endpoint (which is logically distinct from "/builds").

## Testing

Covered by CI.

### CODE REVIEW

File: Gemfile
Comment (twometresteve) [inline Gemfile:3]: Needs updating before merge, once MR is released.

**Review by fractalwrench — APPROVED** [review]
LGTM after the mazerunner target has been updated in the Gemfile

**Review by Cawllec — APPROVED** [review]
LGTM, pending the MR release

**Pull request merged**
